### ISSUE TITLE
MemoryError after several descriptions requiring NVDA restart

### ISSUE THREAD

**Issue by zersiax** [cartertemm/AI-content-describer#26]
After a few, usually 3 or 4, descriptions I get the following traceback:

```python
ERROR - logHandler._threadExceptHook (23:59:00.111) - Thread-235 (describe_image) (82428):
Exception in thread Thread-235 (describe_image):
Traceback (most recent call last):
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "C:\Users\flori\AppData\Roaming\nvda\addons\AIContentDescriber\globalPlugins\AIContentDescriber\__init__.py", line 351, in describe_image
    message = service.process(file, **ch.config[service.name])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\flori\AppData\Roaming\nvda\addons\AIContentDescriber\globalPlugins\AIContentDescriber\description_service.py", line 333, in process
    cache.read_cache()
  File "C:\Users\flori\AppData\Roaming\nvda\addons\AIContentDescriber\globalPlugins\AIContentDescriber\cache.py", line 30, in read_cache
    cache = json.load(f)
            ^^^^^^^^^^^^
  File "json\__init__.pyc", line 293, in load
MemoryError

```

I have turned off all the caching logic, the image modification and this still happens. I'm not sure if NVDA lets users allocate more memory to the NVDA python interpreter, or if I need to delete some kind of cache file but this has been happening on two machines now. Any idea what this might be?

**Comment by cartertemm** [comment]
Hello @zersiax,

Thank you for this report, it's a good catch!

I believe I have fixed it in 0fb524a4066ef4f15141c2c4881a2deee447ccc2. The problem was that we were reading the cache, irrespective of whether that option was enabled in settings.

A good future improvement would be to split up the json file such that it doesn't ever get too big. In the meantime, you can find it at %appdata%\nvda\images.cache.

Let me know if this worked for you.

**Comment by zersiax** [comment]
Yup, I found that file and it had grown to over  200 megs. Deleting it solved the issue.
